### PR TITLE
Add Azure support

### LINKED ISSUE #93: Azure support
- [ ] Implement basic TF for Azure
- [ ] Implement basic AKS scripting for Azure
- [ ] Implement challenge 9 for Azure
- [ ] Implement challenge 10 for Azure
- [ ] Implement challenge 11 for Azure

### PR DESCRIPTION
**Pull request by bendehaan**
Thank you for submitting a pull request to the WrongSecrets app!

Closes #93

### CODE REVIEW

File: src/main/resources/explanations/challenge9_hint-azure.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+You can solve this challenge by the following steps:
+
+1. Dig through terraform.tfstate with Trufflehog:
+- After you have build the infrastructure following the instructions in the README of the `gcp` folder, stay in that folder and run `trufflehog3 --output=terraform-infra.json --format=json .`
```
Comment (commjoen) [inline src/main/resources/explanations/challenge9_hint-azure.adoc:4]:
```suggestion
- After you have build the infrastructure following the instructions in the README of the `azure` folder, stay in that folder and run `trufflehog3 --output=terraform-infra.json --format=json .`
```

File: src/main/java/org/owasp/wrongsecrets/challenges/cloud/Challenge11.java [hunk 2]
```diff
@@ -139,4 +176,12 @@ private String getGCPChallenge11Value() {
         }
         return gcpDefaultValue;
     }
+
+    private String getAzureChallenge11Value() {
+        if (isAzure()) {
+            log.info(String.format("Using Azure Key Vault URI: %s", azureVaultUri));
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/cloud/Challenge11.java:182]: if it is the default value, maybe log that it is being skipped as it does not work then?

File: src/main/resources/explanations/challenge11-azure.adoc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+=== Azure Key Vault part 3
+
+In our AKS cluster, there are two Azure Identity bindings. These are used by Azure to inject metadata. In our case, we use these to access Azure Key Vault.
+
+The interesting thing about key vault is that it sets permissions per vault, rather than per secret. The developers unfortunately only have one key vault for two identities...
+
+Hint: There's a binding to `separate-workload-pod-id`. Try creating a pod with the right labels to steal wrongsecret-3 (see also[https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity#run-a-sample-application]).
```
Comment (commjoen) [inline src/main/resources/explanations/challenge11-azure.adoc:7]:
```suggestion
Hint: There's a binding to `separate-workload-pod-id`. Try creating a pod with the right labels to steal wrongsecret-3 (see also https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity#run-a-sample-application[the aks documentation]).
```

**Review by commjoen — CHANGES_REQUESTED** [review]
Great starting point! Have a few review comments and will test it further tomorrow.
BTW: already refactored java a little so it would have less repetition, but not sure about the name of the method introdcued.

File: src/main/java/org/owasp/wrongsecrets/challenges/cloud/Challenge11.java (hunk 2 above)
Comment (bendehaan) [reply src/main/java/org/owasp/wrongsecrets/challenges/cloud/Challenge11.java:171]: Addressed!

**Review by commjoen — APPROVED** [review]
LGTM! seriously good enough for an RC1 release :) of 1.3.0

**Pull request merged**